test/server/graphql/v2/mutation/MemberMutations.test.ts

```diff
@@ -0,0 +1,268 @@
+          },
+          collectiveAdminUser,
+        );
+
+        if (role === roles.HOST) {
+          expect(result.errors).to.have.length(1);
+          expect(result.errors[0].message).to.equal('You can only edit accountants, admins, or members.');
+        } else {
+          expect(result.errors).to.not.exist;
+          expect(result.data.editMember.role).to.equal(roles.ADMIN);
+          expect(result.data.editMember.description).to.equal(`role changed to ${role}`);
+        }
+      });
+    });
+  });
+
+  describe('removeMember', () => {
+    const removeMemberMutation = gqlV2/* GraphQL */ `
+      mutation RemoveMember(
+        $memberAccount: AccountReferenceInput!
+        $account: AccountReferenceInput!
+        $role: MemberRole!
+        $isInvitation: Boolean
+      ) {
+        removeMember(memberAccount: $memberAccount, account: $account, role: $role, isInvitation: $isInvitation)
+      }
+    `;
+    it('should remove a member from the collective and document the change in an activity', async () => {
+      const randomUser = await fakeUser();
+      await collective.addUserWithRole(randomUser, roles.MEMBER);
+      const result = await utils.graphqlQueryV2(
+        removeMemberMutation,
+        {
+          memberAccount: { id: idEncode(randomUser.CollectiveId, IDENTIFIER_TYPES.ACCOUNT) },
+          account: { id: idEncode(collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          description: 'user 2 role changed to ACCOUNTANT',
+          role: roles.MEMBER,
+        },
+        collectiveAdminUser,
+      );
+
+      const activity = await models.Activity.findAll({
+        where: {
+          type: ActivityTypes.COLLECTIVE_CORE_MEMBER_REMOVED,
+          CollectiveId: collective.id,
+        },
+      });
+
+      expect(result.errors).to.not.exist;
+      expect(activity.length).to.equal(1);
+      expect(activity[0].data.memberCollective.id).to.equal(randomUser.CollectiveId);
+      expect(result.data.removeMember).to.equal(true);
+    });
+    it('should remove the invitation (if not accepted yet)', async () => {
+      const randomInvitedUser = await fakeUser();
+      await models.MemberInvitation.invite(collective, {
+        role: roles.MEMBER,
+        description: 'invite user to delete invitation test case',
+        MemberCollectiveId: randomInvitedUser.id,
+        CreatedByUserId: collectiveAdminUser.id,
+      });
+      const result = await utils.graphqlQueryV2(
+        removeMemberMutation,
+        {
+          memberAccount: { id: idEncode(randomInvitedUser.CollectiveId, IDENTIFIER_TYPES.ACCOUNT) },
+          account: { id: idEncode(collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          description: 'invite user to delete invitation test case',
+          role: roles.MEMBER,
+          isInvitation: true,
+        },
+        collectiveAdminUser,
+      );
+
+      expect(result.errors).to.not.exist;
+      expect(result.data.removeMember).to.equal(true);
```

Comment:
As per the test title, we need to make sure that the invitation gets deleted here. Possible solution:

```suggestion
      expect(result.data.removeMember).to.equal(true);
      await invitation.reload({ paranoid: false });
      expect(invitation.deletedAt).to.not.be.null;
```

Reply:
@Betree resolved.